dashboard/app/views/levels/editors/fields/_song_selection.html.haml

```diff
@@ -8,7 +8,7 @@
     Click to
     %a.select_none{href: '#'} not limit song selection
   %p (shift-click or cmd-click to select multiple).
-  = f.select :song_selection, ((@level.song_selection || []).map {|s| songMap[s] } + @level.class.hoc_songs).uniq, {}, {size: 12, multiple: true}
+  = f.select :song_selection, ((@level.song_selection || []).map {|s| songMap[s] }.select {|s| s} + @level.class.hoc_songs).uniq, {}, {size: 12, multiple: true}
   %p Songs currently selected (does not update until level is saved):
 
-  = @level.song_selection ? @level.song_selection.map {|s| songMap[s][0] } : "All songs are available"
+  = @level.song_selection ? @level.song_selection.map {|s| songMap[s]&.[](0) }.select {|s| s} : "All songs are available"
```

Comment:
I know this is haml, which doesn't lend itself to readability, but maybe something a little more verbose like `songMap[s]&.length && songMap[s][0]` might be a little easier to understand? I see the string `&.[` only a few times in our repo (and never to grab the first element of an array)

Reply:
And this one I'll defer pre-existing styles, so changed as well.

I definitely agree that It's a pretty weird looking syntax.